### ISSUE TITLE
Use hydra for args and defaults

### ISSUE THREAD

**Issue by claforte** [ashawkey/stable-dreamfusion#277]
The way we handle arguments and override defaults is messy and confusing. I think it might be cleaner to use hydra instead, similarly to https://github.com/threestudio-project/threestudio.

I might do it myself soon since the defaults overridden here-and-there are hindering my progress. ;-)

@ashawkey what do you think?

**Comment by csawtelle** [comment]
Sorry to pop in here. This would be really helpful!

**Comment by ashawkey** [comment]
I agree. The current option system is quite confusing indeed... If you could help with this it would be really appriciated!
